### PR TITLE
Add group associations to roles

### PR DESCRIPTION
**Pull request by openwithcode**
## Description

With this PR we are adding group association for roles in order to define to which channel and group is associated with.

### CODE REVIEW

File: examples/medmnist/schema.json [hunk 1]
```diff
@@ -2,34 +2,41 @@
...
-	    "isDataConsumer": true
-	},
-	{
-	    "name": "aggregator",
-	    "description": "It aggregates the updates from trainers"
-	}
+        {
+            "name": "trainer",
+            "description": "It consumes the data and trains local model",
+            "isDataConsumer": true
```
Comment (myungjin) [inline examples/medmnist/schema.json:8]: perhaps `groupAssociation` field is missing.

File: examples/medmnist/schema.json [hunk 2]
```diff
@@ -2,34 +2,41 @@
...
+        },
+        {
+            "name": "aggregator",
+            "description": "It aggregates the updates from trainers",
+            "replica": 1
```
Comment (myungjin) [inline examples/medmnist/schema.json:13]: same here: `groupAssociation` field is missing.

File: pkg/openapi/model_role.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022 Cisco Systems, Inc. and its affiliates
+// Copyright 2023 Cisco Systems, Inc. and its affiliates
```
Comment (myungjin) [inline pkg/openapi/model_role.go:1]: The year should be the year that the work was introduced for the first time. So, please keep the year as is.

**Review by myungjin — COMMENTED** [review]
left a couple of small comments. Please consider to address them.

File: examples/medmnist/schema.json (hunk 1 above)
Comment (openwithcode) [reply examples/medmnist/schema.json]: Indeed it was missing, i've added it here as well.

File: examples/medmnist/schema.json (hunk 2 above)
Comment (openwithcode) [reply examples/medmnist/schema.json]: Indeed it was missing, i've added it here as well.

File: pkg/openapi/model_role.go (hunk 3 above)
Comment (openwithcode) [reply pkg/openapi/model_role.go]: You are right @myungjin the year should not change and should remain the year in which this file was created.

**Review by myungjin — COMMENTED** [review]
lgtm. one last thing I'd like to ask is to squash all the commits into one so that we have a clean hash commit record.

**Review by myungjin — APPROVED** [review]

**Pull request merged**